impexp-core/src/main/java/org/citydb/util/CoreConstants.java

```diff
@@ -47,4 +47,6 @@
     public static final String EXPORT_STUB = "exportStub";
     public static final String EXPORT_AS_ADDITIONAL_OBJECT = "additionalObject";
     public static final String UNIQUE_TEXTURE_FILENAME_PREFIX = "tex_";
+
+    public static boolean IS_GUI_MODE = false;
```

Comment:
It seems that all variables in this class should be _constant_. Maybe the `ObjectRegistry` is a better place for hoding this global mode?

Reply:
Yes, I agree, it's not a perfect palce :-) I think we can get rid of this flag completely, so that's why I moved it here temporarily.

Reply:
Good, you want to get rid of this flag in this branch or after merging into master?

Reply:
This flag is only used to decide whether or not to calculate the total number of features affected by the export/delete operation. This calculation uses an `SQL COUNT()` based on the provided filter and, thus, can be quite slow on large databases. In GUI mode, we always apply this calculation because it allows us to render a nice progress bar. In CLI mode we suppress it in favor of a faster export/delete (especially desirable if you embed the CLI in larger processes and workflows).

Getting rid of this flag requires to implement a new option (both in the GUI and the CLI) for the user to decide whether the number of affected features should be calculated. And if it should be calculated, then my original intention for the CLI mode was to also show a nice progress bar (e.g., https://github.com/ctongfei/progressbar). But I skipped this idea for this PR because it would have taken more time to implement.

So, getting rid of the flag requires more work. I am happy to keep the flag where it is for now. What do you think?

Reply:
agree, this can be done in a separate pull request.